chatkit/src/main/java/com/pusher/chatkit/messages/MessageService.kt

```diff
@@ -46,8 +43,10 @@ class MessageService(
 
     @JvmOverloads
     @UsesCoroutines
-    fun messageEvents(messageLimit: Int? = null): ReceiveChannel<RoomSubscription.Event> =
-        broadcast { messageEvents(messageLimit) { offer(it) } }
+    fun messageEvents(messageLimit: Int? = null): Promise<Result<Message, Error>> =
+        Promise.promise {
```

Comment:
What's this doing?

Reply:
`Promise` is like a `Future` but more flexible and works on old Android. I'm only using it for one-time requests as so far I'm only loading non-stream data, so this may change. I'm unsure if we should have two different types for one time results and subscriptions.